packages/grid/src/vaadin-grid-column-group.js

```diff
@@ -313,10 +312,15 @@ class GridColumnGroup extends ColumnBaseMixin(PolymerElement) {
         info.addedNodes.filter(this._isColumnElement).length > 0 ||
         info.removedNodes.filter(this._isColumnElement).length > 0
       ) {
-        this._preventHiddenCascade = true;
+        // Prevent synchronization of the hidden state to child columns.
+        // If the group is currently auto-hidden, and a visible column is added,
+        // we don't want the other columns to become visible as well.
+        this._preventHiddenSynchronization = true;
         this._rootColumns = this._getChildColumns(this);
         this._childColumns = this._rootColumns;
-        this._preventHiddenCascade = false;
+        this._updateVisibleChildColumns(this._childColumns);
+        this._updateAutoHidden();
```

Comment:
nit: These two functions seem quite interdependent.
- `_updateAutoHidden` needs an up-to-date list of `this._visibleChildColumns` and therefor `_updateVisibleChildColumns` is always called right before `_updateAutoHidden`.
- Calling `_updateVisibleChildColumns` without calling `_updateAutoHidden` right after could potentially lead to an illegal state.

Should we move the invocation of `_updateAutoHidden()` inside `_updateVisibleChildColumns` to make sure it always gets run after updating the visible child columns?